modules/core/shared/src/main/scala/io/circe/Printer.scala

```diff
@@ -334,4 +334,47 @@ final object Printer {
       }
     }
   }
+
+  private[circe] final class AppendableByteBuffer(cs: Charset) extends Appendable {
+    private[this] var index = 0
+    private[this] var chars = new Array[Char](32)
+
+    private[this] def ensureToFit(n: Int): Array[Char] = {
```

Comment:
Is there a reason not to have this return `Unit` and then work directly with `chars`? I don't have a strong opinion here, but I feel like that would be a little more straightforward about where the mutation is happening.